onnxruntime/core/framework/session_options.h

```diff
@@ -64,6 +64,10 @@ struct SessionOptions {
   // See class 'OrtValuePatternPlanner'.
   bool enable_mem_pattern = true;
 
+  bool enable_mem_reuse = true;
+
+  bool transfer_ownership_intermediate_output_tensors = false;
```

Comment:
I am thinking if this can be inferred from the edge connection from TrainingGraph. If BreakOp's input is not consumed by any latter node, then the ownership can be transferred to pytorch. If the input NodeArg is still needed by latter graph, then we need to make a copy of the tensor before handing off to pytorch.

In another word, we need assume that pytorch would override the buffer inplace, thus we should assume transfer_ownership_intermediate_output_tensors is always true. 

---
In reply to: [600916158](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600916158) [](ancestors = 600916158)